### PR TITLE
Dịch "sgd_vn" - Phần 1

### LINKED ISSUE #1824: Dịch "sgd_vn" - Phần 1
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_optimization/sgd_vn.md#L1-L116

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by nguyenduydu**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #1824

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_optimization/sgd_vn.md [hunk 1]
```diff
@@ -54,15 +55,16 @@ If gradient descent is used, the computing cost for each independent variable it
 Therefore, when the model training dataset is large, the cost of gradient descent for each iteration will be very high.
```
Comment (rootonchair) [inline chapter_optimization/sgd_vn.md:55]: Bạn dịch thiếu nhé

File: chapter_optimization/sgd_vn.md [hunk 2]
```diff
@@ -54,15 +55,16 @@ If gradient descent is used, the computing cost for each independent variable it
...
+Nếu hạ gradient được sử dụng, chi phí tính toán cho mỗi vòng lặp độc lập là $\mathcal{O}(n)$ sẽ tăng một cách tuyến tính với $n$.
 
 <!--
 Stochastic gradient descent (SGD) reduces computational cost at each iteration.
 At each iteration of stochastic gradient descent, we uniformly sample an index $i\in\{1,\ldots, n\}$ for data instances at random, 
 and compute the gradient $\nabla f_i(\mathbf{x})$ to update $\mathbf{x}$:
 -->
 
-*dịch đoạn phía trên*
+Hạ gradient ngẫu nhiên (_stochastic gradient descent_ - SGD) sẽ giảm chi phí tính toán ở mỗi vòng lặp.
```
Comment (rootonchair) [inline chapter_optimization/sgd_vn.md:66]:
```suggestion
Hạ gradient ngẫu nhiên (_stochastic gradient descent_ - SGD) giúp giảm chi phí tính toán ở mỗi vòng lặp.
```

File: chapter_optimization/sgd_vn.md [hunk 3]
```diff
@@ -74,7 +76,9 @@ We can see that the computing cost for each iteration drops from $\mathcal{O}(n)
 We should mention that the stochastic gradient $\nabla f_i(\mathbf{x})$ is the unbiased estimate of gradient $\nabla f(\mathbf{x})$.
 -->
 
-*dịch đoạn phía trên*
+Ở đây, $\eta$ là tốc độ học.
+Ta có thể thấy rằng chi phí tính toán cho mỗi vòng lặp giảm từ $\mathcal{O}(n)$ của hạ gradient xuống còn hằng số $\mathcal{O}(1)$.
+Nên nhớ rằng gradient ngẫu nhiên $\nabla f_i(\mathbf{x})$ là một ước lượng không thiên vị của gradient $\nabla f(\mathbf{x})$.
```
Comment (rootonchair) [inline chapter_optimization/sgd_vn.md:81]:
```suggestion
Nên nhớ rằng gradient ngẫu nhiên $\nabla f_i(\mathbf{x})$ là một ước lượng không thiên lệch của gradient $\nabla f(\mathbf{x})$.
```

File: chapter_optimization/sgd_vn.md [hunk 4]
```diff
@@ -84,13 +88,13 @@ $$\mathbb{E}_i \nabla f_i(\mathbf{x}) = \frac{1}{n} \sum_{i = 1}^n \nabla f_i(\m
 This means that, on average, the stochastic gradient is a good estimate of the gradient.
 -->
 
-*dịch đoạn phía trên*
+Do đó, tính trung bình thì gradient ngẫu nhiên là một ước lượng tốt của gradient.
```
Comment (rootonchair) [inline chapter_optimization/sgd_vn.md:91]:
```suggestion
Do đó, về trung bình, gradient ngẫu nhiên là một ước lượng tốt của gradient.
```

File: chapter_optimization/sgd_vn.md (hunk 1 above)
Comment (nguyenduydu) [reply chapter_optimization/sgd_vn.md:55]: Cám ơn em, anh sẽ bổ sung thêm.

**Comment by nguyenduydu** [comment]
Cám ơn @rootonchair  và @minhduc0711 đã góp ý nhé.
